packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -426,7 +426,7 @@ export const useField = <
       // On multi input range pickers we want to update selection range only for the active input
       // This helps to avoid the focus jumping on Safari https://github.com/mui/mui-x/issues/9003
       // because WebKit implements the `setSelectionRange` based on the spec: https://bugs.webkit.org/show_bug.cgi?id=224425
-      if (inputRef.current === getActiveElement(document)) {
+      if (inputRef.current === getActiveElement(inputRef.current.ownerDocument)) {
```

Comment:
Could you use `ownerDocument` util instead?
https://github.com/mui/mui-x/blob/4295a7f78ef992293834c95f8ce80286b17e0558/packages/x-date-pickers/src/internals/components/PickersPopper.tsx#L184